### PR TITLE
[ADD] specific: rename_custom_field

### PR DESCRIPTION
**Pull request by Pirols**
Joining the `rename_custom_{model,module,table,column}` family.

```
> rg 'rename_field.*custom"' src/upgrade-specific/scripts -tpy | wc -l
85
```

Recent example: https://github.com/odoo/upgrade-specific/pull/16967

### CODE REVIEW

**Comment by KangOl** [comment]
upgradeci retry with always only base

File: src/util/specific.py [hunk 1]
```diff
@@ -62,6 +63,21 @@ def rename_custom_model(cr, model_name, new_model_name, custom_module=None, repo
     )
 
 
+def rename_custom_field(cr, old_field_name, new_field_name, model, report_details=""):
```
Comment (UemuS) [inline src/util/specific.py:66]: The entire family would want `model` as the first arg :p

**Review by UemuS — APPROVED** [review]

**Comment by jjmaksoud** [comment]
I have the impression we are discouraged from renaming fields in specific fixes, renaming columns is the way to go unless it doesnt work for some reason

**Comment by Pirols** [comment]
> I have the impression we are discouraged from renaming fields in specific fixes, renaming columns is the way to go unless it doesnt work for some reason

I think it's not much different from renaming a model: if an old custom field at some point clashes with a newly introduced standard field we need to rename or remove it. Renaming the column only wouldn't cut, would it?

**Comment by aj-fuentes** [comment]
I'm not in favor of this. Renaming fields is rarely the way to go for specific fixes. When it is, namely when there is a client's support ticket, it can be done with the existing tools.

**Comment by aj-fuentes** [comment]
> Renaming the column only wouldn't cut, would it?

It does in most cases. If it doesn't then it needs further analysis. But renaming a field is a too big operation that should be avoided.

**Comment by aj-fuentes** [comment]
>  think it's not much different from renaming a model

In some cases we can rename a table and it works. When not then we need to actually rename the model. In most situations where models clash the existing model is a backport of the to-be-added model, or something compatible. Since models have higher chances to be incompatible than single fields, it makes more sense to fully rename a model. But I think renaming a field can be avoided more easily.

**Comment by Pirols** [comment]
IRL discussion summary: While not having this unfortunately won't prevent anyone from renaming fields in specific scripts at least it won't reinforce/validate it more.

**Pull request closed without merging**